Validation/EventGenerator/plugins/LheWeightValidation.cc

```diff
@@ -0,0 +1,336 @@
+#include "Validation/EventGenerator/interface/LheWeightValidation.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "Validation/EventGenerator/interface/DQMHelper.h"
+#include "Validation/EventGenerator/interface/HepMCValidationHelper.h"
+
+using namespace edm;
+
+LheWeightValidation::LheWeightValidation(const edm::ParameterSet& iPSet) {
+  lheLabel_ = iPSet.getParameter<edm::InputTag>("lheProduct"),
+  genParticleToken_ = consumes<reco::GenParticleCollection>(iPSet.getParameter<edm::InputTag>("genParticles"));
+  lheEvtToken_ = consumes<LHEEventProduct>(lheLabel_);
+  lheRunToken_ = consumes<LHERunInfoProduct, edm::InRun>(lheLabel_);
+  genJetToken_ = consumes<reco::GenJetCollection>(iPSet.getParameter<edm::InputTag>("genJets"));
+  dumpLHEheader_ = iPSet.getParameter<bool>("dumpLHEheader");
+  nScaleVar_ = iPSet.getParameter<int>("nScaleVar");
+  idxPdfStart_ = iPSet.getParameter<int>("idxPdfStart");
+  idxPdfEnd_ = iPSet.getParameter<int>("idxPdfEnd");
+  leadLepPtRange_ = iPSet.getParameter<double>("leadLepPtRange");
+  leadLepPtNbin_ = iPSet.getParameter<int>("leadLepPtNbin");
+  leadLepPtCut_ = iPSet.getParameter<double>("leadLepPtCut");
+  lepEtaCut_ = iPSet.getParameter<double>("lepEtaCut");
+  rapidityRange_ = iPSet.getParameter<double>("rapidityRange");
+  rapidityNbin_ = iPSet.getParameter<int>("rapidityNbin");
+  jetPtCut_ = iPSet.getParameter<double>("jetPtCut");
+  jetEtaCut_ = iPSet.getParameter<double>("jetEtaCut");
+  nJetsNbin_ = iPSet.getParameter<int>("nJetsNbin");
+  jetPtRange_ = iPSet.getParameter<double>("jetPtRange");
+  jetPtNbin_ = iPSet.getParameter<int>("jetPtNbin");
+
+  nPdfVar_ = idxPdfEnd_ - idxPdfStart_ + 1;
+}
+
+LheWeightValidation::~LheWeightValidation() {}
+
+void LheWeightValidation::bookHistograms(DQMStore::IBooker& iBook, edm::Run const& iRun, edm::EventSetup const&) {
+  // check LHE product exists
+  edm::Handle<LHERunInfoProduct> lheInfo;
+  iRun.getByLabel(lheLabel_, lheInfo);
+
+  if (!lheInfo.isValid())
+    return;
+
+  ///Setting the DQM top directories
+  std::string folderName = "Generator/LHEWeight";
+  dqm_ = new DQMHelper(&iBook);
```

Comment:
Please move to a smart pointer (e.g. `unique_ptr`)
Or, probably better: do not have dqm_ as class member, make it local and pass it by reference to the `bookTemplate` method.